semantic-core/src/Data/Loc.hs

```diff
@@ -11,22 +13,27 @@ module Data.Loc
 , runFailWithLoc
 ) where
 
-import Control.Applicative
-import Control.Effect.Carrier
-import Control.Effect.Error
-import Control.Effect.Fail
-import Control.Effect.Reader
-import Data.Text (Text, pack)
-import Data.Text.Prettyprint.Doc (Pretty (..))
-import GHC.Stack
-import Prelude hiding (fail)
+import           Control.Applicative
+import           Control.Effect.Carrier
+import           Control.Effect.Error
+import           Control.Effect.Fail
+import           Control.Effect.Reader
+import           Data.Text (Text, pack)
+import           Data.Text.Prettyprint.Doc (Pretty (..))
+import           GHC.Stack
+import           Prelude hiding (fail)
+import qualified TreeSitter.Span as TreeSitter
 
 data Loc = Loc
   { locPath :: !Text
   , locSpan :: {-# UNPACK #-} !Span
   }
   deriving (Eq, Ord, Show)
 
+fromTSSpan :: TreeSitter.Span -> Loc
+fromTSSpan (TreeSitter.Span (TreeSitter.Pos a b) (TreeSitter.Pos c d))
+  = Loc mempty (Span (Pos a b) (Pos c d))
```

Comment:
This `mempty` is due to the fact that we don’t have any filename info. I think the right place to take care of supplying a filename is in the `Compile` class, whose result monad should have a `Reader FilePath` constraint.

Reply:
If we’re going to make `semantic-core` depend on `tree-sitter`, then we might as well just define an `Unmarshal` instance for `Loc` and not have to copy at all.

Otherwise, let’s move this into the compiler instead and remove the dependency.

Reply:
Re: the `mempty`, we could also change this to map `TreeSitter.Span` into core’s `Span`, and then just pass in the `Text` & construct the `Loc` in the caller (`locate`, presumably).

Reply:
Yeah, I think that’s probably what I’ll end up doing.